### ISSUE TITLE
MST over DisplayPort (daisy chain) bug in 6.0.2 and 6.0.3 on HP Dev One

### ISSUE THREAD

**Issue by leviport** [pop-os/linux#187]
It looks like there was a small regression on HP Dev One with our 6.0.2 kernel, and it is still present with the 6.0.3 kernel that's currently in master staging. This regression causes only one of two daisy chained monitors to turn on and work, even though display settings shows both displays, and I can move windows to the turned-off display. Before I log in, if I unplug and re-plug the USB-C cable while I'm on the GDM3 screen, that brings the gray GDM background up on both of the daisy chained monitors, and logging in after that gives me a session with two monitors working just fine.

I also tested some Ubuntu mainline kernel versions (6.0.1, 6.0.3, 6.0.4, and 6.0.6), and they seem to work as expected. That's what is leading me to believe that this is mostly related to the kernel, even though GDM3 and Xorg are definitely playing their own small parts.

Also, Wayland seems a lot less likely to have this problem. If I choose either the COSMIC DE session or the Pop on Wayland session, both external monitors usually turn on and work without issue.

One last note: when I disable MST on the upstream DP monitor, so that the downstream daisy chained display can only display a duplicate of the upstream display, it seems to work every time without issue. Just wanted to include that finding, in case it points to any more useful clues.

**Comment by ozdreamern** [comment]
not related to this?  https://mrk.sh/linux/displaylink/fix-displaylink-kernel6.0/

**Comment by leviport** [comment]
@ozdreamern Thanks for the link, but I don't believe so. DisplayLink != DisplayPort MST.

**Comment by frezbo** [comment]
Cross posting from https://github.com/pop-os/pop/issues/2632

## Problem

Have two external monitors connected via USB C and the second monitor connected to the DP out of the first one. This seemingly broke with some kernel update (I was traveling and came back to find the issue).

The monitors gets detected during the system boot and when system asks for the decryption password and also at the login screen. After logging into PoP! OS both the external displays go blank and shows no signal detected, but the displays option in settings do show the extra monitors. Trying to change the resolution at times might trigger the external monitors to work. I have tried a live PoP OS and the external displays work fine in live mode. So I'd assume this has to do with some process/something that;s specific to PoP OS login process.

## System info

```bash
❯ uname -a
Linux deepwater 6.0.6-76060006-generic #202210290932~1667401208~22.04~d2df702 SMP PREEMPT_DYNAMIC Wed N x86_64 x86_64 x86_64 GNU/Linux

❯ xrandr --verbose
Screen 0: minimum 320 x 200, current 5760 x 1080, maximum 16384 x 16384
eDP connected primary 1920x1080+0+0 (0x59) normal (normal left inverted right x axis y axis) 309mm x 173mm
	Identifier: 0x52
	Timestamp:  792916
	Subpixel:   unknown
	Gamma:      1.0:1.0:1.0
	Brightness: 1.0
	Clones:    
	CRTC:       0
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	_MUTTER_PRESENTATION_OUTPUT: 0 
	EDID: 
		00ffffffffffff000dae0a1400000000
		291d0104a51f11780328659759548e27
		1e505400000001010101010101010101
		010101010101363680a0703820403020
		a60035ad10000018000000fe004e3134
		304843412d4541450a20000000fe0043
		4d4e0a202020202020202020000000fe
		004e3134304843412d4541450a200002
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 
		0 1 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	vrr_capable: 0 
		range: (0, 1)
	abm level: 0 
		range: (0, 4)
	max bpc: 16 
		range: (8, 16)
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 78 
		supported: 78
	non-desktop: 0 
		range: (0, 1)
  1920x1080 (0x59) 138.780MHz -HSync -VSync *current +preferred
        h: width  1920 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height 1080 start 1090 end 1096 total 1112           clock  60.00Hz
  1680x1050 (0x5a) 138.780MHz -HSync -VSync
        h: width  1680 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height 1050 start 1090 end 1096 total 1112           clock  60.00Hz
  1280x1024 (0x5b) 138.780MHz -HSync -VSync
        h: width  1280 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height 1024 start 1090 end 1096 total 1112           clock  60.00Hz
  1440x900 (0x5c) 138.780MHz -HSync -VSync
        h: width  1440 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height  900 start 1090 end 1096 total 1112           clock  60.00Hz
  1280x800 (0x5d) 138.780MHz -HSync -VSync
        h: width  1280 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height  800 start 1090 end 1096 total 1112           clock  60.00Hz
  1280x720 (0x5e) 138.780MHz -HSync -VSync
        h: width  1280 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height  720 start 1090 end 1096 total 1112           clock  60.00Hz
  1024x768 (0x5f) 138.780MHz -HSync -VSync
        h: width  1024 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height  768 start 1090 end 1096 total 1112           clock  60.00Hz
  800x600 (0x60) 138.780MHz -HSync -VSync
        h: width   800 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height  600 start 1090 end 1096 total 1112           clock  60.00Hz
  640x480 (0x61) 138.780MHz -HSync -VSync
        h: width   640 start 1968 end 2000 total 2080 skew    0 clock  66.72KHz
        v: height  480 start 1090 end 1096 total 1112           clock  60.00Hz
HDMI-A-0 disconnected (normal left inverted right x axis y axis)
	Identifier: 0x53
	Timestamp:  792916
	Subpixel:   unknown
	Clones:    
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	vrr_capable: 0 
		range: (0, 1)
	max bpc: 8 
		range: (8, 16)
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 85 
		supported: 85
	non-desktop: 0 
		range: (0, 1)
DisplayPort-0 disconnected (normal left inverted right x axis y axis)
	Identifier: 0x54
	Timestamp:  792916
	Subpixel:   unknown
	Clones:    
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	subconnector: Unknown 
		supported: Unknown, VGA, DVI-D, HDMI, DP, Wireless, Native
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	vrr_capable: 0 
		range: (0, 1)
	max bpc: 8 
		range: (8, 16)
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 89 
		supported: 89
	non-desktop: 0 
		range: (0, 1)
DisplayPort-1 disconnected (normal left inverted right x axis y axis)
	Identifier: 0x55
	Timestamp:  792916
	Subpixel:   unknown
	Clones:    
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	subconnector: Unknown 
		supported: Unknown, VGA, DVI-D, HDMI, DP, Wireless, Native
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	vrr_capable: 0 
		range: (0, 1)
	max bpc: 8 
		range: (8, 16)
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 95 
		supported: 95
	non-desktop: 0 
		range: (0, 1)
DisplayPort-2 disconnected (normal left inverted right x axis y axis)
	Identifier: 0x56
	Timestamp:  792916
	Subpixel:   unknown
	Clones:    
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	_MUTTER_PRESENTATION_OUTPUT: 0 
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 
		0 1 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	max bpc: 8 
		range: (8, 16)
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 102 
		supported: 102
	non-desktop: 0 
		range: (0, 1)
DisplayPort-3 disconnected (normal left inverted right x axis y axis)
	Identifier: 0x57
	Timestamp:  792916
	Subpixel:   unknown
	Clones:    
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	_MUTTER_PRESENTATION_OUTPUT: 0 
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 
		0 1 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	max bpc: 8 
		range: (8, 16)
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 105 
		supported: 105
	non-desktop: 0 
		range: (0, 1)
DisplayPort-4 connected 1920x1080+1920+0 (0x62) normal (normal left inverted right x axis y axis) 598mm x 336mm
	Identifier: 0x705
	Timestamp:  792916
	Subpixel:   unknown
	Gamma:      1.0:1.0:1.0
	Brightness: 1.0
	Clones:    
	CRTC:       1
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	_MUTTER_PRESENTATION_OUTPUT: 0 
	EDID: 
		00ffffffffffff0010ac864157303532
		301d0104a53c22783a4815a756529c27
		0f5054a54b00714f8180a9c0d1c00101
		010101010101023a801871382d40582c
		450056502100001e000000ff00443159
		5a5353320a2020202020000000fc0044
		454c4c20503237313948430a000000fd
		00384c1e5311010a20202020202000b5
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 
		0 1 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	max bpc: 8 
		range: (8, 16)
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 120 
		supported: 120
	non-desktop: 0 
		range: (0, 1)
  1920x1080 (0x62) 148.500MHz +HSync +VSync *current +preferred
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  67.50KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  60.00Hz
  1600x900 (0x63) 108.000MHz +HSync +VSync
        h: width  1600 start 1624 end 1704 total 1800 skew    0 clock  60.00KHz
        v: height  900 start  901 end  904 total 1000           clock  60.00Hz
  1280x1024 (0x64) 135.000MHz +HSync +VSync
        h: width  1280 start 1296 end 1440 total 1688 skew    0 clock  79.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  75.02Hz
  1280x1024 (0x65) 108.000MHz +HSync +VSync
        h: width  1280 start 1328 end 1440 total 1688 skew    0 clock  63.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  60.02Hz
  1152x864 (0x66) 108.000MHz +HSync +VSync
        h: width  1152 start 1216 end 1344 total 1600 skew    0 clock  67.50KHz
        v: height  864 start  865 end  868 total  900           clock  75.00Hz
  1024x768 (0x67) 78.750MHz +HSync +VSync
        h: width  1024 start 1040 end 1136 total 1312 skew    0 clock  60.02KHz
        v: height  768 start  769 end  772 total  800           clock  75.03Hz
  1024x768 (0x68) 65.000MHz -HSync -VSync
        h: width  1024 start 1048 end 1184 total 1344 skew    0 clock  48.36KHz
        v: height  768 start  771 end  777 total  806           clock  60.00Hz
  800x600 (0x69) 49.500MHz +HSync +VSync
        h: width   800 start  816 end  896 total 1056 skew    0 clock  46.88KHz
        v: height  600 start  601 end  604 total  625           clock  75.00Hz
  800x600 (0x6a) 40.000MHz +HSync +VSync
        h: width   800 start  840 end  968 total 1056 skew    0 clock  37.88KHz
        v: height  600 start  601 end  605 total  628           clock  60.32Hz
  640x480 (0x6b) 31.500MHz -HSync -VSync
        h: width   640 start  656 end  720 total  840 skew    0 clock  37.50KHz
        v: height  480 start  481 end  484 total  500           clock  75.00Hz
  640x480 (0x6c) 25.175MHz -HSync -VSync
        h: width   640 start  656 end  752 total  800 skew    0 clock  31.47KHz
        v: height  480 start  490 end  492 total  525           clock  59.94Hz
  720x400 (0x6d) 28.320MHz -HSync +VSync
        h: width   720 start  738 end  846 total  900 skew    0 clock  31.47KHz
        v: height  400 start  412 end  414 total  449           clock  70.08Hz
DisplayPort-5 connected 1920x1080+3840+0 (0x62) normal (normal left inverted right x axis y axis) 598mm x 336mm
	Identifier: 0x706
	Timestamp:  792916
	Subpixel:   unknown
	Gamma:      1.0:1.0:1.0
	Brightness: 1.0
	Clones:    
	CRTC:       2
	CRTCs:      0 1 2 3
	Transform:  1.000000 0.000000 0.000000
	            0.000000 1.000000 0.000000
	            0.000000 0.000000 1.000000
	           filter: 
	_MUTTER_PRESENTATION_OUTPUT: 0 
	EDID: 
		00ffffffffffff0010ac844142545342
		0c1e0104a53c22783a4815a756529c27
		0f5054a54b00714f8180a9c0d1c00101
		010101010101023a801871382d40582c
		450056502100001e000000ff00374b51
		5a5653320a2020202020000000fc0044
		454c4c205032373139480a20000000fd
		00384c1e5311010a20202020202000b8
	GAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	DEGAMMA_LUT_SIZE: 4096 
		range: (0, -1)
	GAMMA_LUT: 0 
		range: (0, 65535)
	CTM: 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 
		0 1 
	DEGAMMA_LUT: 0 
		range: (0, 65535)
	TearFree: auto 
		supported: off, on, auto
	max bpc: 8 
		range: (8, 16)
	HDCP Content Type: HDCP Type0 
		supported: HDCP Type0, HDCP Type1
	Content Protection: Undesired 
		supported: Undesired, Desired, Enabled
	underscan vborder: 0 
		range: (0, 128)
	underscan hborder: 0 
		range: (0, 128)
	underscan: off 
		supported: off, on, auto
	scaling mode: None 
		supported: None, Full, Center, Full aspect
	link-status: Good 
		supported: Good, Bad
	CONNECTOR_ID: 117 
		supported: 117
	non-desktop: 0 
		range: (0, 1)
  1920x1080 (0x62) 148.500MHz +HSync +VSync *current +preferred
        h: width  1920 start 2008 end 2052 total 2200 skew    0 clock  67.50KHz
        v: height 1080 start 1084 end 1089 total 1125           clock  60.00Hz
  1600x900 (0x63) 108.000MHz +HSync +VSync
        h: width  1600 start 1624 end 1704 total 1800 skew    0 clock  60.00KHz
        v: height  900 start  901 end  904 total 1000           clock  60.00Hz
  1280x1024 (0x64) 135.000MHz +HSync +VSync
        h: width  1280 start 1296 end 1440 total 1688 skew    0 clock  79.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  75.02Hz
  1280x1024 (0x65) 108.000MHz +HSync +VSync
        h: width  1280 start 1328 end 1440 total 1688 skew    0 clock  63.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  60.02Hz
  1152x864 (0x66) 108.000MHz +HSync +VSync
        h: width  1152 start 1216 end 1344 total 1600 skew    0 clock  67.50KHz
        v: height  864 start  865 end  868 total  900           clock  75.00Hz
  1024x768 (0x67) 78.750MHz +HSync +VSync
        h: width  1024 start 1040 end 1136 total 1312 skew    0 clock  60.02KHz
        v: height  768 start  769 end  772 total  800           clock  75.03Hz
  1024x768 (0x68) 65.000MHz -HSync -VSync
        h: width  1024 start 1048 end 1184 total 1344 skew    0 clock  48.36KHz
        v: height  768 start  771 end  777 total  806           clock  60.00Hz
  800x600 (0x69) 49.500MHz +HSync +VSync
        h: width   800 start  816 end  896 total 1056 skew    0 clock  46.88KHz
        v: height  600 start  601 end  604 total  625           clock  75.00Hz
  800x600 (0x6a) 40.000MHz +HSync +VSync
        h: width   800 start  840 end  968 total 1056 skew    0 clock  37.88KHz
        v: height  600 start  601 end  605 total  628           clock  60.32Hz
  640x480 (0x6b) 31.500MHz -HSync -VSync
        h: width   640 start  656 end  720 total  840 skew    0 clock  37.50KHz
        v: height  480 start  481 end  484 total  500           clock  75.00Hz
  640x480 (0x6c) 25.175MHz -HSync -VSync
        h: width   640 start  656 end  752 total  800 skew    0 clock  31.47KHz
        v: height  480 start  490 end  492 total  525           clock  59.94Hz
  720x400 (0x6d) 28.320MHz -HSync +VSync
        h: width   720 start  738 end  846 total  900 skew    0 clock  31.47KHz
        v: height  400 start  412 end  414 total  449           clock  70.08Hz
```

this is a ThinkPad T14 with AMD integrated CPU/GPU

## Debug Update

 switching to wayland seems to have fixed this, so it seems it's something of a combination between Xorg and the kernel version probably.

**Comment by frezbo** [comment]
Okay quick update:

Updating to the latest kernel `6.0.6-76060006-generic` (got an update today) fixed the issue with Xorg. I guess the ThinkPad issue can be considered resolved now.

**Comment by frezbo** [comment]
> Okay quick update:
> 
> Updating to the latest kernel `6.0.6-76060006-generic` (got an update today) fixed the issue with Xorg. I guess the ThinkPad issue can be considered resolved now.

Seems this is not fully resolved, if the screen is locked and unlocked the external displays goes back to a blank state again but still shows the displays as detected in display settings.

**Comment by leviport** [comment]
Yeah I wouldn't expect that update to fix this. That was a small fix for something unrelated.

**Comment by leviport** [comment]
I believe this will be resolved by 6.0.12: https://github.com/pop-os/linux/pull/194

**Comment by frezbo** [comment]
Seems to be working after an upgrade and reboot, if issue persists I'll report back. Thanks for the great work.
